tests/test_db.py

```diff
@@ -186,6 +234,16 @@ def test_no_error_when_in_transaction(self) -> None:
 
         assert code_is_reached is True
 
+    @_parametrize_transaction_testcase
+    def test_different_database(self) -> None:
+        """
+        Transactions on other databases do not fulfill the requirement.
+        """
+        with pytest.raises(db._MissingRequiredTransaction):  # noqa: SLF001
+            with db.transaction(using=DEFAULT):
```

Comment:
I would swap these so it's clear that the error is coming from `transaction_required`, not `transaction`.